src/main/java/hudson/plugins/favorite/filter/FavoriteFilter.java

```diff
@@ -23,15 +22,10 @@ public FavoriteFilter() {
     @Override
     public List<TopLevelItem> filter(List<TopLevelItem> added, List<TopLevelItem> all, View filteringView) {
         List<TopLevelItem> filtered = new ArrayList<>();
-
-        Authentication authentication = Hudson.getAuthentication();
-
+        Authentication authentication = Jenkins.getAuthentication2();
```

Comment:
To be consistent with all other places I would suggest to use `User.current()` which already ensures that when not null, the user is logged in and when null, access happens as anonymous.